cmake/pika_add_compile_test.cmake

```diff
@@ -5,7 +5,7 @@
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 function(pika_add_compile_test category name)
-  set(options FAILURE_EXPECTED NOLIBS OBJECT)
+  set(options FAILURE_EXPECTED NOLIBS OBJECT COMPILE_ONLY)
```

Comment:
Are not all tests added through `pika_add_compile_test` `COMPILE_ONLY`? In other words, I don't think you need the option here. Then you can revert the explicit flags you've added in the modules.